Assets/HoloToolkit/SpatialMapping/Scripts/RemoteMapping/MeshSaver.cs

```diff
@@ -0,0 +1,282 @@
+                byte[] data = new byte[stream.Length];
+                stream.Read(data, 0, data.Length);
+
+                meshes.AddRange(SimpleMeshSerializer.Deserialize(data));
+            }
+
+            Debug.Log("Mesh file loaded.");
+
+            return meshes;
+        }
+
+        /// <summary>
+        /// Opens the specified file for reading.
+        /// </summary>
+        /// <param name="folderName">The name of the folder containing the file.</param>
+        /// <param name="fileName">The name of the file, including extension. </param>
+        /// <returns>Stream used for reading the file's data.</returns>
+        private static Stream OpenFileForRead(string folderName, string fileName)
+        {
+            Stream stream = null;
+
+#if !UNITY_EDITOR && UNITY_WSA
+            Task<Task> task = Task<Task>.Factory.StartNew(
+                            async () =>
+                            {
+                                StorageFolder folder = await StorageFolder.GetFolderFromPathAsync(folderName);
+                                StorageFile file = await folder.GetFileAsync(fileName);
+                                IRandomAccessStreamWithContentType randomAccessStream = await file.OpenReadAsync();
+                                stream = randomAccessStream.AsStreamForRead();
+                            });
+            task.Wait();
+            task.Result.Wait();
+#else
+            stream = new FileStream(Path.Combine(folderName, fileName), FileMode.Open, FileAccess.Read);
+#endif
+            return stream;
+        }
+
+        /// <summary>
+        /// Opens the specified file for writing.
+        /// </summary>
+        /// <param name="folderName">The name of the folder containing the file.</param>
+        /// <param name="fileName">The name of the file, including extension.</param>
+        /// <returns>Stream used for writing the file's data.</returns>
+        /// <remarks>If the specified file already exists, it will be overwritten.</remarks>
+        private static Stream OpenFileForWrite(string folderName, string fileName)
+        {
+            Stream stream = null;
+
+#if !UNITY_EDITOR && UNITY_WSA
+            Task<Task> task = Task<Task>.Factory.StartNew(
+                            async () =>
+                            {
+                                StorageFolder folder = await StorageFolder.GetFolderFromPathAsync(folderName);
+                                StorageFile file = await folder.CreateFileAsync(fileName, CreationCollisionOption.ReplaceExisting);
+                                IRandomAccessStream randomAccessStream = await file.OpenAsync(FileAccessMode.ReadWrite);
+                                stream = randomAccessStream.AsStreamForWrite();
+                            });
+            task.Wait();
+            task.Result.Wait();
+#else
+            stream = new FileStream(Path.Combine(folderName, fileName), FileMode.Create, FileAccess.Write);
```

Comment:
Should we make this an enumeration if this blocks the main thread?
If we don't care, then why use async for WSA?